rostime/include/ros/duration.h

```diff
@@ -90,8 +100,12 @@ class DurationBase
   bool operator>=(const T &rhs) const;
   bool operator<=(const T &rhs) const;
   double toSec() const { return static_cast<double>(sec) + 1e-9*static_cast<double>(nsec); };
-  int64_t toNSec() const {return static_cast<int64_t>(sec)*1000000000ll + static_cast<int64_t>(nsec);  };
+  double toMSec() const { return static_cast<double>(sec * MSecInSec) + 1e-6*static_cast<double>(nsec); };
+  double toUSec() const { return static_cast<double>(sec * USecInSec) + 1e-3*static_cast<double>(nsec); };
```

Comment:
These two methods returning a `double` are example which would loose precision and therefore should be avoided imo.

Reply:
They lose precision the same way `double toSec()` does. Do you suggest not providing these methods at all is better than providing them in a way that may lead to precision loss? Generally, I think that if someone converts a custom data type to double, he has to expect some kind of precision loss. With these methods, it's pretty obvious to me.

Reply:
I assume that is what @tfoote is worried about and raise the concern in the original issue: https://github.com/ros/roscpp_core/issues/114#issuecomment-583202691

Maybe he can comment and clarify.